sql/rpl_master_info_file.hh

```diff
@@ -0,0 +1,613 @@
+/*
+  Copyright (c) 2025 MariaDB
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; version 2 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along
+  with this program; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin St, Fifth Floor, Boston, MA 02110-1335 USA.
+*/
+
+#include "rpl_info_file.hh"
+#include <unordered_map> // Type of @ref MasterInfoFile::FIELDS_MAP
+#include <string_view>   // Key type of @ref MasterInfoFile::FIELDS_MAP
+#include <unordered_set> // Parameter type of ChangeMaster::set_defaults() //?
+#include <optional>      // Storage type of @ref OptionalIntField
+#include "sql_const.h"   // MAX_PASSWORD_LENGTH
+
+
+inline static constexpr uint32_t SLAVE_MAX_HEARTBEAT_PERIOD=
+  std::numeric_limits<uint32_t>::max() / 1000;
+
+/**
+  A three-way comparison function for using
+  sort_dynamic() and bsearch() on IDArrayField::array.
+  @return -1 if first argument is less, 0 if it equal to, or 1 if it is greater
+  than the second
+  @deprecated Use a sorted set, such as @ref std::set,
+  to save on explicitly calling those functions.
+*/
+inline static int change_master_id_cmp(const void *arg1, const void *arg2)
+{
+  const ulong &id1= *(const ulong *)arg1, &id2= *(const ulong *)arg2;
+  return (id2 > id1) - (id2 < id1);
+}
+
+/// enum for @ref MasterInfoFile::master_use_gtid
+enum struct enum_master_use_gtid { NO, CURRENT_POS, SLAVE_POS, DEFAULT };
+/// String names for non-@ref enum_master_use_gtid::DEFAULT values
+inline const char *master_use_gtid_names[]=
+  {"No", "Current_Pos", "Slave_Pos", nullptr};
+
+/**
+  `mariadbd` Options for the `DEFAULT` values of @ref MasterInfoFile fields
+  @{
+*/
+/// Computes the `DEFAULT` value of @ref ::master_heartbeat_period
+extern uint slave_net_timeout;
+inline uint32_t master_connect_retry= 60;
+inline std::optional<uint32_t> master_heartbeat_period= std::nullopt;
+inline bool master_ssl= true;
+inline const char *master_ssl_ca     = "";
+inline const char *master_ssl_capath = "";
+inline const char *master_ssl_cert   = "";
+inline const char *master_ssl_crl    = "";
+inline const char *master_ssl_crlpath= "";
+inline const char *master_ssl_key    = "";
+inline const char *master_ssl_cipher = "";
+inline bool master_ssl_verify_server_cert= true;
+/// `ulong` is the data type `my_getopt` expects.
+inline auto master_use_gtid= static_cast<ulong>(enum_master_use_gtid::DEFAULT);
+inline uint64_t master_retry_count= 100'000;
+/// }@
+
+
+struct MasterInfoFile: InfoFile
```

Comment:
Name conflict, I know.
But, `Master_info` & `Relay_log_info` also appear to contain runtime data for IO and SQL threads, respectively.
This inadequacy in organization makes maintenance difficult to know what belongs to whom, not to mention one of the 0b10 hardest problems in software development.